webapp/app/.server/service/proposal-times.ts

```diff
@@ -0,0 +1,41 @@
+import { PROPOSAL_STATES, type StatusTime } from "@/utils/proposal-states";
+
+export function daysInSeconds(days: number): number {
+  return days * 24 * 3600;
+}
+
+export function calculateStatusPendingDays(
+  status: PROPOSAL_STATES,
+  creationTimestamp: number,
+  guardiansExtendedLegalVeto: boolean,
+  nowInSeconds: number
+): StatusTime | null {
+  if (status === PROPOSAL_STATES.LegalVetoPeriod) {
+    const delta = nowInSeconds - creationTimestamp;
+    const currentDay = Math.ceil(delta / daysInSeconds(1));
+    const totalDays = guardiansExtendedLegalVeto ? 7 : 3;
+
+    return {
+      totalDays: totalDays,
+      currentDay: currentDay,
+    };
+  }
+
+  if (status === PROPOSAL_STATES.Waiting) {
+    const days3 = daysInSeconds(3);
+    const days7 = daysInSeconds(7);
+    const vetoPeriodDuration = guardiansExtendedLegalVeto ? days7 : days3;
+    const delta = nowInSeconds - (creationTimestamp + vetoPeriodDuration);
+    const currentDay = Math.ceil(delta / daysInSeconds(1));
+    return {
+      totalDays: 30,
```

Comment:
where does 30 come from?

Reply:
I took this from the contract. In the docs is documented as `Risk Review Period`. I removed to it's own better named and explained constant.